images/virtualization-artifact/pkg/controller/service/size_policy_service_test.go

```diff
@@ -0,0 +1,696 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service_test
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/deckhouse/virtualization-controller/pkg/controller/service"
+	v1alpha2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+var vmClassValues map[string]client.Object
+
+var _ = Describe("Spec policy comlience service", func() {
+	var mock *service.ClientMock
+	var ctx context.Context
+
+	BeforeEach(func() {
+		mock = &service.ClientMock{}
+		mock.GetFunc = func(_ context.Context, key client.ObjectKey, obj client.Object, _ ...client.GetOption) error {
+			val, ok := vmClassValues[key.Name]
+			if !ok {
+				return fmt.Errorf("Object not found")
+			}
+
+			obj.(*v1alpha2.VirtualMachineClass).Spec.SizingPolicies = val.(*v1alpha2.VirtualMachineClass).Spec.SizingPolicies
+
+			return nil
+		}
+		vmClassValues = make(map[string]client.Object)
+	})
+
+	Context("testing mock no value", func() {
+		It("Should not value", func() {
+			val := v1alpha2.VirtualMachineClass{}
+			err := mock.Get(ctx, types.NamespacedName{
+				Name: "vmclasstest",
+			}, &val)
+			Expect(err).ShouldNot(BeNil())
+		})
+	})
+
+	Context("testing mock value", func() {
+		BeforeEach(func() {
+			vmClassValues["vmclasstest"] = &v1alpha2.VirtualMachineClass{
+				Spec: v1alpha2.VirtualMachineClassSpec{
+					SizingPolicies: make([]v1alpha2.SizingPolicy, 0),
+				},
+			}
+			vmClassValues["vmclasstest"].(*v1alpha2.VirtualMachineClass).Spec.SizingPolicies = append(
+				vmClassValues["vmclasstest"].(*v1alpha2.VirtualMachineClass).Spec.SizingPolicies,
+				v1alpha2.SizingPolicy{
+					Cores: &v1alpha2.SizingPolicyCores{
+						Min: 1,
+						Max: 4,
+					},
+					Memory: &v1alpha2.SizingPolicyMemory{
+						MemoryMinMax: v1alpha2.MemoryMinMax{
+							Min: resource.MustParse("1Gi"),
+							Max: resource.MustParse("4Gi"),
+						},
+					},
+				},
+			)
+		})
+
+		It("Should value", func() {
+			val := v1alpha2.VirtualMachineClass{}
+			err := mock.Get(ctx, types.NamespacedName{
+				Name: "vmclasstest",
+			}, &val)
+			Expect(err).Should(BeNil())
+			Expect(len(val.Spec.SizingPolicies)).Should(Equal(1))
+		})
+	})
+
+	Context("Vm with no class", func() {
```

Comment:
Use VM or VirtualMachine instead of Vm. vm is the abbreviation, either all letters are uppercase, or none

Reply:
VirtualMachine without class